lib/MusicBrainz/Server/Controller/EditNote.pm

```diff
@@ -0,0 +1,208 @@
+
+    unless ($is_own_note || $is_admin) {
+      $c->stash(
+          component_path  => 'user/UserMessage',
+          component_props => {
+              title    => l('Cannot modify edit note'),
+              message  => l('You cannot modify other users’ edit notes'),
+          },
+          current_view    => 'Node',
+      );
+      $c->detach;
+    }
+
+    my $is_note_deleted = $edit_note->{latest_change} &&
+                          $edit_note->{latest_change}->{status} eq 'deleted';
+
+    if ($is_note_deleted && !$is_admin) {
+      $c->stash(
+          component_path  => 'user/UserMessage',
+          component_props => {
+              title    => l('Cannot modify edit note'),
+              message  => l('This note has already been deleted.'),
+          },
+          current_view    => 'Node',
+      );
+      $c->detach;
+    }
+
+    my $all_edit_notes = $edit->{edit_notes};
+    my $note_index = first {
+      @$all_edit_notes[$_]->id == $edit_note->id
+    } 0..@$all_edit_notes;
+    my @after_note = splice(@$all_edit_notes, $note_index + 1);
+    my $has_reply = 0;
+    for (@after_note) {
+      if ($_->editor_id != $edit_note->editor_id){
+        $has_reply = 1;
+      }
+    }
+
+    if ($has_reply && !$is_admin) {
+      $c->stash(
+          component_path  => 'user/UserMessage',
+          component_props => {
+              title    => l('Cannot modify edit note'),
+              message  => l('You can’t modify this, since somebody else has already replied to it.'),
+          },
+          current_view    => 'Node',
+      );
+      $c->detach;
+    }
+
+    my $now = DateTime->now( time_zone => $edit_note->post_time->time_zone );
+    my $note_age = $now - $edit_note->post_time;
+    my $is_note_too_old = $note_age->{days} > 0;
+
+    if ($is_note_too_old && !$is_admin) {
+      $c->stash(
+          component_path  => 'user/UserMessage',
+          component_props => {
+              title    => l('Cannot modify edit note'),
+              message  => l(
+                'This note was made too long ago. If there’s an important reason why you need it removed, please {contact_url|contact us}.',
+                {contact_url => $CONTACT_URL},
+              ),
+          },
+          current_view    => 'Node',
+      );
+      $c->detach;
+    }
+}
+
+sub delete : PathPart('delete') Chained('load') {
+    my ($self, $c) = @_;
+
+    my $form = $c->form( form => 'EditNoteDelete' );
+    my $edit_note = $c->stash->{edit_note};
+    my $edit_id = $edit_note->{edit_id};
+    my $edit = $c->model('Edit')->get_by_id($edit_id);
+    load_everything_for_edits($c, [ $edit ]);
+
+    $self->detach_if_cannot_modify($c, $edit_note, $edit);
+
+    if ($c->form_posted_and_valid($form)) {
+        if ($form->field('cancel')->input) {
+          # Do nothing
+        } else {
+          $c->model('MB')->with_transaction(sub {
+            $c->model('EditNote')->remove_content(
+              $edit_note,
```

Comment:
A lot of the indentation in this file is two spaces instead of four

Reply:
Oh, can change. That said, is there a reason we generally do 2 spaces in JS and 4 in Perl?